wire/internal/wire/wire_test.go

```diff
@@ -284,40 +279,64 @@ func isIdent(s string) bool {
 }
 
 type testCase struct {
-	name       string
-	pkg        string
-	goFiles    map[string][]byte
-	wantOutput []byte
-
-	wantError        bool
-	wantErrorStrings []string
+	name                 string
+	pkg                  string
+	goFiles              map[string][]byte
+	wantProgramOutput    []byte
+	wantWireOutput       []byte
+	wantWireError        bool
+	wantWireErrorStrings []string
 }
 
 // loadTestCase reads a test case from a directory.
 //
 // The directory structure is:
 //
 //	root/
-//		pkg        file containing the package name containing the inject function
-//		           (must also be package main)
-//		out.txt    file containing the expected output, or starting with the
-//		           magic line "ERROR" if this test should cause generation to
-//		           fail (any subsequent lines are substrings that should
-//		           appear in the errors).
-//		...        any Go files found recursively placed under GOPATH/src/...
+//
+//		pkg
+//			file containing the package name containing the inject function
+//			(must also be package main)
+//
+//		...
+//			any Go files found recursively placed under GOPATH/src/...
+//
+//		want/
+//
+//			wire_errs.txt
+//					expected errors from the Wire Generate function,
+//					missing if no errors expected
+//
+//			wire_gen.go
+//					verified output of wire from a test run with
+//					-record, missing if wire_errs.txt is present
+//
+//			program_out.txt
+//					expected output from the final compiled program,
+//					missing if wire_errs.txt is present
+//
 func loadTestCase(root string, wireGoSrc []byte) (*testCase, error) {
 	name := filepath.Base(root)
 	pkg, err := ioutil.ReadFile(filepath.Join(root, "pkg"))
 	if err != nil {
 		return nil, fmt.Errorf("load test case %s: %v", name, err)
 	}
-	out, err := ioutil.ReadFile(filepath.Join(root, "out.txt"))
-	if err != nil {
-		return nil, fmt.Errorf("load test case %s: %v", name, err)
-	}
-	wantErrorStrings, wantError := parseGoldenOutput(out)
-	if wantError {
-		out = nil
+	var wantProgramOutput []byte
+	var wantWireOutput []byte
+	wireErrb, err := ioutil.ReadFile(filepath.Join(root, "want", "wire_errs.txt"))
+	wantWireError := err == nil
+	var wantWireErrorStrings []string
+	if wantWireError {
+		wantWireErrorStrings = strings.Split(strings.TrimSpace(string(wireErrb)), "\n")
+	} else {
+		wantWireOutput, err = ioutil.ReadFile(filepath.Join(root, "want", "wire_gen.go"))
+		if err != nil {
+			return nil, fmt.Errorf("load test case %s: %v. If this is a new testcase, run with -record to generate the wire_gen.go file.", name, err)
```

Comment:
Wouldn't this fail everything new under record?

Reply:
Good point. Fixed it.